pkg/reconciler/common/releases.go

```diff
@@ -291,10 +294,23 @@ func allReleases(instance v1alpha1.KComponent) ([]string, error) {
 
 // latestRelease returns the latest release tag available under kodata directory for Knative component.
 func latestRelease(instance v1alpha1.KComponent) string {
+	return getLatestRelease(instance, "")
+}
+
+// getLatestRelease returns the latest release tag available under kodata directory for Knative component
+// based on spec.version.
+func getLatestRelease(instance v1alpha1.KComponent, version string) string {
+	// The versions are in a descending order, so the first one will be the latest version.
 	vers, err := allReleases(instance)
 	if err != nil {
 		panic(err)
 	}
-	// The versions are in a descending order, so the first one will be the latest version.
-	return vers[0]
+	for _, val := range vers {
+		if strings.HasPrefix(val, version) {
```

Comment:
I think here we want version + '.' or we'll match incorrectly in some cases

for ex searching for version 1 might match 11 if there isn't an actual 1.0

Reply:
oh, that's correct.